front/app/components/EsriMap/index.tsx

```diff
@@ -0,0 +1,131 @@
+import Fullscreen from '@arcgis/core/widgets/Fullscreen';
+
+// hooks
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+
+// utils
+import { getTileAttribution } from './utils';
+import { isNil } from 'utils/helperUtils';
+import { DEFAULT_TILE_PROVIDER } from './types';
+
+type Props = {
+  center?: GeoJSON.Point | null;
+  height?: string;
+  width?: string;
+  zoom?: number;
+  maxZoom?: number;
+  layers?: Layer[];
+  onClick?: (event: any, mapView: MapView) => void;
+  setLocationPoint?: (locationPoint: GeoJSON.Point) => void;
+  locationPoint?: GeoJSON.Point;
+  graphics?: Graphic[];
+  showFullscreenOption?: boolean;
+};
+
+const EsriMap = ({
+  center,
+  zoom,
+  maxZoom,
+  layers,
+  onClick,
+  height,
+  width,
+  graphics,
+  showFullscreenOption,
+}: Props) => {
+  const { data: appConfig } = useAppConfiguration();
+
+  useEffect(() => {
+    // Get global map settings from the app config
+    const globalMapSettings = appConfig?.data.attributes.settings.maps;
+
+    // Create a new map and map view
+    const esriMap = new Map();
+
+    const mapView = new MapView({
+      container: 'esriMap', // Reference to DOM node that will contain the view
+      map: esriMap,
+      zoom: zoom || globalMapSettings?.zoom_level,
+      center: !isNil(center)
+        ? [center.coordinates[0], center.coordinates[1]]
+        : undefined,
+      constraints: {
+        maxZoom: maxZoom || 22,
+        minZoom: 8,
+      },
+    });
+
+    // Set the basemap
+    const webTileLayerFromUrl = new WebTileLayer({
+      urlTemplate: globalMapSettings?.tile_provider || DEFAULT_TILE_PROVIDER,
+      copyright: getTileAttribution(globalMapSettings?.tile_provider || ''),
+    });
+
+    esriMap.basemap = new Basemap({
+      baseLayers: [webTileLayerFromUrl],
+    });
+
+    // Add any layers that were passed in
+    if (layers) {
+      layers.forEach((layer) => {
+        esriMap.add(layer);
+      });
+    }
+
+    // Add any graphics that were passed in. These should sit on top of any map layers.
+    if (graphics) {
+      graphics.forEach((graphic) => {
+        mapView.graphics.add(graphic);
+      });
+    }
+
+    // Add fullscreen widget if set
+    if (showFullscreenOption) {
+      const fullscreen = new Fullscreen({
+        view: mapView,
+      });
+      mapView.ui.add(fullscreen, 'top-right');
+    }
+
+    // On map click, pass the event to onClick handler if it was provided
+    mapView.on('click', function (event) {
+      // By passing the mapview to onClick functions, we can easily change the map from that function
+      onClick && onClick(event, mapView);
+    });
+  }, [
+    appConfig?.data.attributes.settings.maps,
+    center,
+    graphics,
+    layers,
+    maxZoom,
+    onClick,
+    showFullscreenOption,
+    zoom,
+  ]);
+
+  return (
+    <>
+      <link
+        rel="stylesheet"
+        href="https://js.arcgis.com/4.28/esri/themes/light/main.css"
```

Comment:
I'm not quite sure how best to add this in? Is it alright to add the <link> directly like this?

Reply:
It's probably better to add it like so ` import '@arcgis/core/assets/esri/themes/light/main.css' ` at the top of the file?